src/components/data_search/DatasetSearchTable.js

```diff
@@ -40,40 +43,121 @@ export const DatasetSearchTable = (props) => {
+              "fields": [
+                  "datasetName",
+                  "dataLocation",
+                  "study.description",
+                  "study.studyName",
+                  "study.species",
+                  "study.piName",
+                  "study.dataCustodianEmail",
+                  "study.dataTypes",
+                  "dataUse.primary.code",
+                  "dataUse.secondary.code",
+                  "dac.dacName",
+                  "datasetIdentifier"
+              ]
+          }
+      }
+      ];
+      queryChunks.push(...searchModifier);
+    }
+
+    var filterQuery = {};
+    if (filters.length > 0) {
+      const shouldTerms = [];
+    
+      filters.forEach(term => {
+        shouldTerms.push({
+          "term": {
+            "accessManagement": term
+          }
+        });
+      });
+    
+      if (shouldTerms.length > 0) {
+        filterQuery = [
+          {
+            "bool": {
+              "should": shouldTerms
+            }
+          }
+        ];
+      }
+    }
+    
+    // do not add filter subquery if no filters are applied
+    if (filters.length > 0) {
+      return {
+        'from': 0,
+        'size': 10000,
+        'query': {
+          'bool': {
+            'must': queryChunks,
+            'filter': filterQuery
+          }
+        }
+      };
+    } else {
+      return {
+        'from': 0,
+        'size': 10000,
+        'query': {
+          'bool': {
+            'must': queryChunks
+          }
+        }
+      };
+    }
+  };
+
+  const filterHandler = (event, data, filter, searchTerm) => {
     var newFilters = [];
-    if (!isFiltered(filter)) {
+    if (!isFiltered(filter) && filter !== '') {
       newFilters = filters.concat(filter);
     } else {
       newFilters = filters.filter((f) => f !== filter);
     }
     setFilters(newFilters);
 
-    var newFiltered = [];
-    if (newFilters.length === 0) {
-      newFiltered = data;
-    } else {
-      newFiltered = data.filter((dataset) => {
-        // TODO: remove extra checks when openAccess property is deprecated
-        if (newFilters.includes('open') && (dataset.openAccess || dataset.accessManagement === 'open')) {
-          return true;
-        }
-        if (newFilters.includes('controlled') && (
-          (!dataset.openAccess && dataset.accessManagement === undefined) || (dataset.openAccess === undefined && dataset.accessManagement === 'controlled')
-        )) {
-          return true;
-        }
-        if (newFilters.includes('external') && dataset.accessManagement === 'external') {
-          return true;
-        }
-        return false;
-      });
-    }
-    setFiltered(newFiltered);
+    const fullQuery = assembleFullQuery(searchTerm, newFilters);
+    const search = async () => {
+      try {
+        await DataSet.searchDatasetIndex(fullQuery).then((filteredDatasets) => {
+          setFiltered(datasets.filter(value => filteredDatasets.some(item => isEqual(item, value))));
```

Comment:
```suggestion
          var newFiltered = datasets.filter(value => filteredDatasets.some(item => isEqual(item, value)));
          setFiltered(newFiltered);
```

nit: we can inspect the results a bit more easily this way.